src/citrine/resources/file_link.py

```diff
@@ -289,3 +297,30 @@ def _complete_upload(self, dest_name: str, uploader: _Uploader):
 
         url = self._get_path(file_id) + '/versions/{}'.format(version)
         return FileLink(filename=dest_name, url=url)
+
+    def download(self, file_link: FileLink, local_path: str):
+        """
+        Download the file associated with a given FileLink to the local computer.
+
+        Parameters
+        ----------
+        file_link: FileLink
+            Resource referencing the external file.
+        local_path: str
+            Path to save file on the local computer. If `local_path` is a directory,
+            then the filename of this FileLink object will be appended to the path.
+
+        """
+        directory, filename = os.path.split(local_path)
+        if not filename:
+            filename = file_link.filename
+        if not os.path.isdir(directory):
```

Comment:
If my `cwd` is `/home/maxhutch/` and `directory=/home/maxhutch/file_test/0/1/2` but there is no `/home/maxhutch/file_test/0/1/2`, what does `isdir` return?

Reply:
False.

It looks to me like `isdir` considers both absolute paths and paths from the cwd. So if that directory _did_ exist you could get there either with `directory=/home/maxhutch/file_test/0/1/2` or with `directory=file_test/0/1/2`

Reply:
But `open` only uses absolute path, so I should add a check that it's a valid absolute path to a directory.

Reply:
I take that back. Both `open` and `isdir` cue off of starting with a backslash (or the equivalent cue in Windows) to determine if it is a relative or absolute path. So this should work.